### DISCUSSION TITLE
Is it possible to force it to only parse object like values?

### DISCUSSION THREAD

**Discussion by babakfp** [eemeli/yaml#531] [Q&A] [answered]
Hi 👋

Can I configure it to throw an error whenever it receives values other than:

```yaml
---
message: Hello, World!
---
```

Meaning if the value was empty, `null`, `undefined`, array `[]`, object `{}`, `false`, `true` etc, throw an error and say that only the above syntax is valid.

**Comment by eemeli** [comment]
Have you tried the `{ schema: 'failsafe' }` option? It won't throw errors, but it'll treat all scalar values as strings.

**Reply by babakfp** [reply]
I think I explained it wrong. I want to only allow (key value) syntax as top level value. And the nested values can be whatever.

```yaml
name: GitHub
todos:
    one: Hi
    two: Bye
```

**Reply by eemeli** [reply, marked as answer]
In that case you'll probably need to implement a custom validator of some sort for your data, and apply it to the parsed results.

**Reply by babakfp** [reply]
Ok, thanks.
